pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -344,7 +346,7 @@ private boolean isNoDictionaryColumn(Set<String> noDictionaryColumns, Set<String
       // if the column is part of noDictionary set from table config
       if (fieldSpec instanceof DimensionFieldSpec && _aggregateMetrics && (dataType == FieldSpec.DataType.STRING ||
           dataType == FieldSpec.DataType.BYTES)) {
-        _logger.info("Not creating dictionary in consuming segment for column {} of type {}", column, dataType.toString());
+        _logger.info("Will create dictionary in consuming segment for column {} of type {}", column, dataType.toString());
```

Comment:
Let's be more explicit, to clarify that that noDictionary is being overwritten by aggregateMetrics for the column.

Reply:
done